core/src/main/java/org/elasticsearch/index/translog/TranslogReader.java

```diff
@@ -115,8 +117,9 @@ public static TranslogReader open(
                     case TranslogWriter.VERSION_CHECKSUMS:
                         throw new IllegalStateException("pre-2.0 translog found [" + path + "]");
                     case TranslogWriter.VERSION_CHECKPOINTS:
+                    case TranslogWriter.VERSION_TIMESTAMPS:
                         assert path.getFileName().toString().endsWith(Translog.TRANSLOG_FILE_SUFFIX) : "new file ends with old suffix: " + path;
-                        assert checkpoint.numOps >= 0 : "expected at least 0 operatin but got: " + checkpoint.numOps;
+                        assert checkpoint.numOps >= 0 : "expected at least 0 operation but got: " + checkpoint.numOps;
```

Comment:
just an idea, should we use the last modification time on the file instead? I mean it should be ok to do that no?